scripts/superdesk/editor/spellcheck/spellcheck.js

```diff
@@ -285,19 +297,26 @@ function SpellcheckService($q, api, dictionaries, $rootScope, $location) {
      * @param {string} word
      */
     this.suggest = function suggest(word) {
-        return api.save('spellcheck', {
-            word: word,
-            language_id: lang
-        }).then(function(result) {
-            var allDict = getDict();
-            var wordFoundInDict = _.pick(allDict.content, function(value, key) {
-                if (key.toLowerCase() === word.toLowerCase()) {
-                    return key;
-                }
+        if (word.match(/^\s+$/i)) {
+            return Promise.resolve([{key: ' ', value: 'Add single space'}]);
+        } else {
+            return api.save('spellcheck', {
+                word: word,
+                language_id: lang
+            }).then(function(result) {
+                var allDict = getDict();
+                var wordFoundInDict = _.pick(allDict.content, function(value, key) {
+                    if (key.toLowerCase() === word.toLowerCase()) {
+                        return key;
+                    }
+                });
+                angular.extend(result.corrections, Object.keys(wordFoundInDict));
+
+                return result.corrections.map(function(key) {
```

Comment:
You may use ES6 syntax here if you'd like `return result.corrections.map(key => ({key: key, value: key}));`